### PR TITLE
functionality for zero effort projects

### PR DESCRIPTION
**Pull request by Abhishek25081999**
Targets #2095

### Description
Weekly alert to Key Account Managers about zero expected team effort.

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Project/Emails/ZeroEffortInProjectMail.php [hunk 1]
```diff
@@ -29,10 +29,12 @@ public function __construct(array $projectManager)
...
-        return $this
-        ->to($this->projectManager['email'])
-        ->subject('ColoredCow Portal - Some of your team members have zero effort in projects!')
-        ->view('project::mail.zero-effort-team-member-list')
-        ->with(['projectManager' => $this->projectManager]);
+        $mail = $this->subject('ColoredCow Portal - Some of your team members have zero effort in projects!');
+        foreach ($this->projectDetail as $project) {
+            $mail->to($project['email']);
+        }
+
```
Comment (gautam071) [inline Modules/Project/Emails/ZeroEffortInProjectMail.php:36]: Remove empty space

File: Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php [hunk 2]
```diff
@@ -4,24 +4,28 @@
 			line-height: 1px;
 		}
 	</style>
-	<p>Hello {{ $projectManager['name'] }},</p>
-	<p>We found some projects where the expected hours are zero for you or team members where you are assigned as project manager. Please update these projects:</p>
+     @foreach ($projectDetail as $project)
+		
+	 @endforeach
```
Comment (gautam071) [inline Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php:9]: Why empty for each loop?

File: Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php [hunk 3]
```diff
@@ -4,24 +4,28 @@
...
 			<th scope="col">Project Name</th>
 			</tr>
 		</thead>
 		<tbody>
-			@foreach ($projectManager['projects'] as $project)
 				<tr>
 					<td>
-						<a href="{{ route('project.show', $project) }}">{{ $project->name }}</a>
+						@foreach ($projectDetail as $project)
+
```
Comment (gautam071) [inline Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php:22]: Remove empty space

File: Modules/Project/Console/ZeroEffortInProject.php [hunk 4]
```diff
@@ -40,10 +40,10 @@ public function __construct()
      */
     public function handle()
     {
-        $service = app(ProjectServiceContract::class);
-        $getMailDetailsForProjectManagers = $service->getMailDetailsForProjectManagers();
-        foreach ($getMailDetailsForProjectManagers as $projectManager) {
-            Mail::queue(new ZeroEffortInProjectMail($projectManager));
+        $service = app(ProjectService::class);
```
Comment (gautam071) [inline Modules/Project/Console/ZeroEffortInProject.php:43]: What is the reason for changing from `ProjectServiceContract` to `ProjectService`?

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: Modules/Project/Emails/ZeroEffortInProjectMail.php (hunk 1 above)
Comment (Abhishek25081999) [reply Modules/Project/Emails/ZeroEffortInProjectMail.php:36]: ok

File: Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php (hunk 2 above)
Comment (Abhishek25081999) [reply Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php:9]: ok i'll remove it.

File: Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php (hunk 3 above)
Comment (Abhishek25081999) [reply Modules/Project/Resources/views/mail/zero-effort-team-member-list.blade.php:22]: ok

**Comment by gautam071** [comment]
Looks like earlier suggested feedbacks was not fixed please check that

**Review by gautam071 — CHANGES_REQUESTED** [review]
@Abhishek25081999 
Left some comments and there are some unresolved comments as well

File: Modules/Project/Emails/ZeroEffortInProjectMail.php [hunk 5]
```diff
@@ -10,16 +10,16 @@ class ZeroEffortInProjectMail extends Mailable
...
-    protected $projectManager;
+    public $projectDetail;
 
     /**
      * Create a new message instance.
      *
      * @return void
      */
-    public function __construct(array $projectManager)
+    public function __construct($projectDetail)
```
[the remark this answers is not in the document]
Comment (Abhishek25081999) [reply Modules/Project/Emails/ZeroEffortInProjectMail.php]: @gautam071 Done!

File: Modules/Project/Console/ZeroEffortInProject.php [hunk 6]
```diff
@@ -41,9 +41,9 @@ public function __construct()
     public function handle()
     {
         $service = app(ProjectServiceContract::class);
-        $getMailDetailsForProjectManagers = $service->getMailDetailsForProjectManagers();
-        foreach ($getMailDetailsForProjectManagers as $projectManager) {
-            Mail::queue(new ZeroEffortInProjectMail($projectManager));
+        $projectDetails = $service->getMailDetailsForKeyAccountManager();
+        foreach ($projectDetails as $projectDetail) {
+            Mail::queue(new ZeroEffortInProjectMail($projectDetail));
```
Comment (gautam071) [inline Modules/Project/Console/ZeroEffortInProject.php:46]: Variable name changes why?

File: Modules/Project/Emails/ZeroEffortInProjectMail.php [hunk 7]
```diff
@@ -10,29 +10,29 @@ class ZeroEffortInProjectMail extends Mailable
...
-    protected $projectManager;
+    public $projectDetail;
 
     /**
      * Create a new message instance.
      *
      * @return void
      */
-    public function __construct(array $projectManager)
+    public function __construct(array $projectDetail)
```
Comment (gautam071) [inline Modules/Project/Emails/ZeroEffortInProjectMail.php:20]: here as well?
what's the reason

File: Modules/Project/Services/ProjectService.php [hunk 8]
```diff
@@ -307,34 +306,21 @@ public function saveOrUpdateProjectContract($data, $project)
         }
     }
 
-    public function getMailDetailsForProjectManagers()
+    public function getMailDetailsForKeyAccountManager()
```
Comment (gautam071) [inline Modules/Project/Services/ProjectService.php:309]: The email is going out for multiple managers some 
```suggestion
    public function getMailDetailsForKeyAccountManagers()
```

Comment (gautam071) [reply Modules/Project/Services/ProjectService.php:309]: Also change it every where

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: Modules/Project/Console/ZeroEffortInProject.php (hunk 6 above)
Comment (Abhishek25081999) [reply Modules/Project/Console/ZeroEffortInProject.php:46]: variable name changed after foreach loop.

**Review by gautam071 — APPROVED** [review]
looks good

**Pull request merged**